### PR TITLE
Support `J` in visual block mode

### PR DESCRIPTION
**Pull request by lusingander**
**What this PR does / why we need it**:

`J` (join lines) works in visual block mode in original Vim, but this is not supported this extension.

expected:

image

If `J` is executed in the state of this image, expected to be:

```
one two three
four
```

**Which issue(s) this PR fixes**

**Special notes for your reviewer**:

- Cursor position after execution is different from original Vim, but this is the same behavior as visual mode and visual line mode. So I think this should be fixed as another Pull Request.
  - In the above example, cursor should be before `three` after execution, but actually it is before `two`.
- `gJ` doesn't work either, but doesn't fix it. Because `gJ` has a bug like #2207 and it doesn't work properly.

And if necessary, I'd like to fix and create another PR for these as well 👻

### CODE REVIEW

File: src/actions/commands/actions.ts [hunk 1]
```diff
@@ -3270,6 +3275,24 @@ class ActionJoinVisualMode extends BaseCommand {
...
+class ActionJoinVisualBlockMode extends BaseCommand {
+  modes = [Mode.VisualBlock];
+  keys = ['J'];
+  mightChangeDocument = true;
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    let actionJoin = new ActionJoin();
+    const range = vimState.cursors[0];
+    const topLeft = visualBlockGetTopLeftPosition(range.start, range.stop);
+    const bottomRight = visualBlockGetBottomRightPosition(range.start, range.stop);
```
Comment (J-Fields) [inline src/actions/commands/actions.ts]: Could you use `vimState.cursorStartPosition` and `vimState.cursorStopPosition` instead?

File: src/actions/commands/actions.ts [hunk 2]
```diff
@@ -3270,6 +3275,24 @@ class ActionJoinVisualMode extends BaseCommand {
...
+
+    vimState.currentRegisterMode = RegisterMode.CharacterWise;
+    vimState = await actionJoin.execJoinLines(topLeft, bottomRight, vimState, 1);
```
Comment (J-Fields) [inline src/actions/commands/actions.ts]: Just construct the `ActionJoin` inline `await new ActionJoin().execJoinLines(...);`

File: test/mode/modeVisualBlock.test.ts [hunk 3]
```diff
@@ -188,4 +188,12 @@ suite('Mode Visual Block', () => {
     keysPressed: '<C-v>llj"ayGo<C-r>a<Esc>',
     end: ['abcde', '01234', 'abcde', '01234', '123', 'bcd', '|'],
   });
+
+  newTest({
```
Comment (J-Fields) [inline test/mode/modeVisualBlock.test.ts]: Can you write a test where the entire visual block is on the same line? It's a bit of a special case.

**Comment by J-Fields** [comment]
> I'd like to fix and create another PR for these as well

That'd be fantastic, thanks 🙂

**Comment by lusingander** [comment]
@J-Fields thanks for the confirmation, I fixed these!

File: src/actions/commands/actions.ts [hunk 4]
```diff
@@ -3270,6 +3275,22 @@ class ActionJoinVisualMode extends BaseCommand {
...
+  modes = [Mode.VisualBlock];
+  keys = ['J'];
+  mightChangeDocument = true;
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    const start = vimState.cursorStartPosition;
+    const end = vimState.cursorStopPosition;
+
+    vimState.currentRegisterMode = RegisterMode.CharacterWise;
+    vimState = await new ActionJoin().execJoinLines(start, end, vimState, 1);
```
Comment (J-Fields) [inline src/actions/commands/actions.ts]: Could you condense this like so: `return new ActionJoin().execJoinLines(vimState.cursorStartPosition, vimState.cursorStopPosition, vimState, 1);`?

Sorry to nitpick again; really appreciate the contributions! 🙂

Comment (lusingander) [reply src/actions/commands/actions.ts]: No problem, I think this is good, thank you 😀

**Comment by J-Fields** [comment]
Thanks again!

**Pull request merged**